Sources/NIOSOCKS/Messages/ClientGreeting.swift

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// The SOCKS handshake begins with the client sending a greeting
+/// containing supported authentication methods.
+public struct ClientGreeting: Hashable {
+    
+    /// The SOCKS protocol version - we currently only support v5.
+    public var version: UInt8
```

Comment:
If we only support one version is there any reason for this to be `var`?

Reply:
I was on the fence about this. It's not really needed, but maybe neat to have in there? If you feel strongly I'll just delete it.

Reply:
I don't think it should be `var`, it just causes us issues. What should users do if this says `4`? What should we do?

I'm fine with having it, but having it be `let` or a computed property that returns 5 is probably better.